drivers/serial/pty.c

```diff
@@ -223,22 +223,22 @@ static void pty_destroy(FAR struct pty_devpair_s *devpair)
 {
   char devname[16];
 
-  /* Un-register the slave device */
-
-#ifdef CONFIG_PSEUDOTERM_BSD
-  snprintf(devname, 16, "/dev/ttyp%d", devpair->pp_minor);
-#else
-  snprintf(devname, 16, "/dev/pts/%d", devpair->pp_minor);
-#endif
-  unregister_driver(devname);
-
   /* Un-register the master device (/dev/ptyN may have already been
    * unlinked).
    */
 
   snprintf(devname, 16, "/dev/pty%d", (int)devpair->pp_minor);
```

Comment:
why do we need typecast here, but not below?

Reply:
Done, remove the unnecessary cast.